### PR TITLE
BEI-11: setup ci/cd with github actions

### PR DESCRIPTION
**Pull request by LucianaDera**
WHY THIS PR?
To make it easy to set up and manage CI/CD pipelines directly within our repository.

WHAT DOES THIS PR DO?
-  It allows you to automatically build, test, and validate your code whenever changes are pushed.

HOW CAN THIS PR BE TESTED?
- Create test workflow.
-  Trigger the test workflows by making changes to the test.

THINGS YOU MAY WANT THE REVIEWER TO NOTE.
N/A

### CODE REVIEW

**Comment by hizikex** [comment]
Please resolve the conflicts on this files.

**Comment by LordUche** [comment]
@LucianaDera you should branch out of the default branch which is `staging`.

**Review by RICHYDOS — APPROVED** [review]

**Review by hizikex — APPROVED** [review]

**Comment by LordUche** [comment]
Great work @LucianaDera 👍
Please resolve the merge conflicts

File: tsconfig.json [hunk 1]
```diff
@@ -12,9 +12,9 @@
...
-    "strict": true,
-    "noUnusedLocals": true,
-    "noImplicitReturns": true,
-    "skipLibCheck": true
+    "strict": true, 
+    "noUnusedLocals": true, 
+    "noImplicitReturns": true, 
+    "skipLibCheck": true  
+  },
   }
```
Comment (LordUche) [inline tsconfig.json:20]:
```suggestion
    "strict": true,
    "noUnusedLocals": true,
    "noImplicitReturns": true,
    "skipLibCheck": true
  }
}
```

**Review by LordUche — CHANGES_REQUESTED** [review]

File: superLinter.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14
```
Comment (LordUche) [inline superLinter.yml:22]:
```suggestion
          node-version: 18
```
One more change

File: superLinter.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+name: super-Linter
+
+on:
+  push:
+    branches:
+      - main
```
Comment (LordUche) [inline superLinter.yml:6]:
```suggestion
      - staging
```

File: superLinter.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  pull_request:
+    branches:
+      - main
```
Comment (LordUche) [inline superLinter.yml:9]:
```suggestion
      - staging
```

**Review by LordUche — CHANGES_REQUESTED** [review]

**Pull request merged**